modules/ServerAPI/src/client/SaltBrowser.coffee

```diff
@@ -0,0 +1,402 @@
+		"click .bv_cancelDelete": "handleCancelDeleteClicked"
+		"click .bv_downloadSaltBtn": "handleDownloadSaltClicked"
+		"click .bv_createNewSaltBtn": "handleCreateSaltClicked"
+		"click .bv_confirmCreateSaltButton": "handleConfirmCreateSaltClicked"
+		"click .bv_cancelCreate": "handleCancelCreateClicked"
+		"click .bv_editSalt": "handleEditSaltClicked"
+		"click .bv_confirmEditSaltButton": "handleConfirmEditSaltClicked"
+		"click .bv_cancelEdit":"handleCancelEditSaltClicked"
+
+	initialize: ->
+		template = _.template($("#SaltBrowserView").html())
+		$(@el).empty()
+		$(@el).html template
+		@searchController = new SaltSimpleSearchController
+			model: new SaltSearch()
+			el: @$('.bv_saltSearchController')
+		@searchController.render()
+		@searchController.on "searchReturned", @setupSaltSummaryTable.bind(@)
+		@$('.bv_queryToolDisplayName').html window.conf.service.result.viewer.displayName
+
+	setupSaltSummaryTable: (salts) =>
+		@destroySaltSummaryTable()
+
+		$(".bv_searchingSaltsMessage").addClass "hide"
+		if salts is null
+			@$(".bv_errorOccurredPerformingSearch").removeClass "hide"
+
+		else if salts.length is 0
+			@$(".bv_noMatchingSaltsFoundMessage").removeClass "hide"
+			@$(".bv_saltTableController").html ""
+		else
+			$(".bv_searchSaltsStatusIndicator").addClass "hide"
+			@$(".bv_saltTableController").removeClass "hide"
+			@saltSummaryTable = new SaltSummaryTableController
+				collection: new SaltList salts
+			@saltSummaryTable.on "selectedRowUpdated", @selectedSaltUpdated
+			$(".bv_saltTableController").html @saltSummaryTable.render().el
+
+	selectedSaltUpdated: (salt) =>
+		@trigger "selectedSaltUpdated"
+		@saltController = new SaltEditorController
+			model: new Salt salt.attributes
+			readOnly: true
+
+		$('.bv_saltController').html @saltController.render().el
+		$(".bv_saltController").removeClass("hide")
+		$(".bv_saltControllerContainer").removeClass("hide")
+
+		# Need to Use This Route (Or Similar) to Display Salt Structure
+		# app.get '/api/chemStructure/renderStructureByCode', exports.renderStructureByCode
+
+		# Need to Get 
+		requestJSON = {
+			#	"codeName" :  "#{@saltController.model.get("id")}"
+				"molStructure" : "#{@saltController.model.get("molStructure")}",
+				"height" : 100, 
+				"width" : 100, 
+				"format" : "png"
+			}
+
+
+		$.ajax(
+			type: 'POST'
+			url: "/api/chemStructure/renderMolStructureBase64"
+			contentType: 'application/json'
+			dataType: 'text'
+
+			data: JSON.stringify(requestJSON)
+			success: (base64ImagePNG) => # Rendered Image Should Be Displayed on Sever
+				pngSrc = "data:image/png;base64," + base64ImagePNG
+				pngImage = '<img src="' + pngSrc + '" />'
+				@$('.bv_structureHolder').html pngImage 
+
+				@$('.bv_editSalt').show()
+				if window.conf.salt?.editingRoles?
+					editingRoles = window.conf.salt.editingRoles.split(",")
+					if !UtilityFunctions::testUserHasRole(window.AppLaunchParams.loginUser, editingRoles)
```

Comment:
Don't think we should invent a new role for this.  The server side code uses cmpdreg admin role so I think we should just use that here as well.

Reply:
Just to expand on that, I think cmpdreg admins should be able to both edit and delete, and cmpdreg users should only be able to create and view.

Reply:
Implemented